device/iot-device-client/src/main/java/com/microsoft/azure/sdk/iot/device/transport/amqps/AmqpsIotHubConnection.java

```diff
@@ -421,158 +356,95 @@ private void closeAsync()
-     *
-     * @param event The Proton Event object.
-     */
     @Override
     public void onConnectionInit(Event event)
     {
-        // Codes_SRS_AMQPSIOTHUBCONNECTION_15_025: [The event handler shall get the Connection (Proton) object from the event handler and set the host name on the connection.]
         this.connection = event.getConnection();
-        this.connection.setHostname(this.hostName);
-
-        // Codes_SRS_AMQPSIOTHUBCONNECTION_15_029: [The event handler shall open the connection.]
+        this.connection.setHostname(hostName);
         this.connection.open();
-        try
-        {
-            // Codes_SRS_AMQPSIOTHUBCONNECTION_12_009: [The event handler shall call the amqpsSessionManager.onConnectionInit function with the connection.]
-            this.amqpsSessionManager.onConnectionInit(this.connection);
-        }
-        catch (TransportException e)
+
+        //Create one session per multiplexed device, or just one session if not multiplexing
+        for (AmqpsSessionHandler amqpsSessionHandler : this.sessionHandlerList)
         {
-            this.savedException = e;
-            log.error("Encountered an exception while reacting to onConnectionInit within amqp session", e);
+            amqpsSessionHandler.session = connection.session();
+            amqpsSessionHandler.session.open();
         }
     }
 
-    @Override
-    public void onSessionRemoteOpen(Event e)
-    {
-        this.amqpsSessionManager.onSessionRemoteOpen(e.getSession());
-    }
-
-    /**
-     * Event handler for the connection bound event. Sets Sasl authentication and proper authentication mode.
-     *
-     * @param event The Proton Event object.
-     */
     @Override
     public void onConnectionBound(Event event)
     {
-        // Codes_SRS_AMQPSIOTHUBCONNECTION_15_030: [The event handler shall get the Transport (Proton) object from the event.]
-        Transport transport = event.getConnection().getTransport();
-        if (transport != null)
+        Transport transport = event.getTransport();
+
+        if (this.deviceClientConfig.isUseWebsocket())
         {
-            if (this.useWebSockets)
-            {
-                addWebsocketLayer(transport);
-            }
+            addWebsocketLayer(transport);
+        }
 
-            try
-            {
-                // Codes_SRS_AMQPSIOTHUBCONNECTION_15_031: [The event handler shall call the AmqpsSessionManager.onConnectionBound with the transport and the SSLContext.]
-                this.amqpsSessionManager.onConnectionBound(transport);
-            }
-            catch (TransportException e)
-            {
-                this.savedException = e;
-                log.error("Encountered an exception while reacting to onConnectionBound event within amqp session", this.savedException);
-            }
+        try
+        {
+            this.amqpsAuthenticationLinkHandler.setSslDomain(transport, this.deviceClientConfig.getAuthenticationProvider().getSSLContext());
```

Comment:
Add a note here specifying that this will always be 1 per connection, and that is why it is initialized here (even when multiplexed devices exist).